CONTRIBUTING.md

```diff
@@ -47,6 +47,93 @@ Using commit signoffs tags is mandatory for all commits; we also
+
+Our code style follows [the Google C++ Style
+Guide](https://google.github.io/styleguide/cppguide.html) with a few
+exceptions. To run automatic code style formatting on files you modify, use our
+[clang-format
+template](https://github.com/mendersoftware/mendertesting/blob/master/.clang-format):
+
+```bash
+curl -O https://github.com/mendersoftware/mendertesting/raw/master/.clang-format
+clang-format -i --style=file:.clang-format <MODIFIED_FILES>
+```
+
+Note that the template requires clang-format version 15 or higher.
+
+##### C++ standard
+
+All shared code must use C++ features from no later than the C++11
+standard. Platform code for POSIX platforms is allowed to use C++17 if
+necessary, but C++11 is preferred for consistency with the rest of the code.
+
+#### File names
+
+Files must be named with `.cpp` and `.hpp` extensions. Only use `.h` if the
+header holds C declarations.
+
+##### Indentation
+
+Indentation in Mender code uses 1 tab per level, no spaces. Function arguments
+or initializers that are broken over multiple lines are indented one level.
+
+##### Line length
+
+Our line lengths are capped at 100 characters instead of 80, which is Google's
+cap. When considering line lengths, each tab is considered 4 characters
+wide. [The other guidelines for line
+length](https://google.github.io/styleguide/cppguide.html#Line_Length) still
+apply.
+
+##### Namespaces
+
+All the [rules about
+namespaces](https://google.github.io/styleguide/cppguide.html#Namespaces) apply,
+with one exception: It is allowed to use `using namespace std` to import the
+`std` namespace. All other namespaces must be spelled out where they are used
+(though it is allowed to shorten long names using `namespace short_name =
+long_name`).
+
+##### C headers
+
+In general, C++ libraries and wrappers should be preferred over C
+equivalents. However, it is permitted to include C headers if no good
+alternative is available. This must only be done inside platform code.
+
+If including headers from the Standard C Library, use this form:
+
+```
+#include <cstring>  // Good
+#include <string.h> // Bad, should not be used in C++ code.
+```
+
+##### Code blocks
+
+All code blocks occurring as part of `if` statements, `for` loops or `while`
+loops, must be enclosed in curly brackets, even single line blocks.
+
+##### Class access specifiers
+
+Class access specifiers, `public`, `protected` and `private` must be indented at
+the same level as their parent class, in other words at the same level as where
+the `class` keyword appears.
+
+#### Organization
+
+All code that interacts with the operating system, such as file access, network
+operations, timers, process creation, etc, must happen through a C++
+interface. These interfaces must be organized in a
+`common/<MODULE>/<MODULE>.hpp` hierarchy. The platform implementation must
+reside in `common/<MODULE>/impl/<MODULE>.cpp` files. It is allowed to use more
+files in addition to the `<MODULE>` named files, if necessary.
```

Comment:
What is an example of `<MODULE>` here? And shouldn't the implementations be in `common/<MODULE>/impl/<PLATFORM>.cpp` or something like that?

Reply:
For this part I just followed what was done in https://github.com/mendersoftware/mender/pull/1091, so this is open to change. Let's settle it there first, and return here after.

Reply:
Updated and made an example of what `<MODULE>` can be.

While I agree we might need `<PLATFORM>.cpp` later, I'm leaving this one for later. It won't really come up until Q2 or Q3 next year, at which point we probably need to change many other things we hadn't thought of as well.